sources/launcher/Stride.Launcher/ViewModels/PackageVersionViewModel.cs

```diff
@@ -316,6 +317,38 @@ private async Task DeleteInternal(bool displayErrorMessage)
             }
         }
 
+        private static async Task UninstallPackage(NugetStore store, NugetPackage package, ProgressReport progressReport)
+        {
+            await store.UninstallPackage(package, progressReport);
+
+            // If package is GameStudio, then recursively delete its Stride/Xenko dependencies to save more disk space
+            if (store.MainPackageIds.Contains(package.Id))
+            {
+                await UninstallDependencies(store, package);
+            }
+        }
+
+        private static async Task UninstallDependencies(NugetStore store, NugetPackage package)
+        {
+            foreach (var dependency in package.Dependencies)
+            {
+                string dependencyId = dependency.Item1;
+                string dependencyIdPrefix = dependencyId.Split('.').First();
+                PackageVersionRange dependencyVersionRange = dependency.Item2;
+
+                // Dependency must be from Stride/Xenko and package version must match exactly
```

Comment:
Good idea, it should prevent it from uninstalling the shared `Stride.Samples.Templates` (which is versioned separately and usually shared amongst multiple versions).

Reply:
Thinking about it, I am not sure how this will behave if at some point we have versions that overlap and evolve at different speeds, because the `dependencyVersionRange` is usually a "at least" constraint, not a "equal" (i.e. if both 4.1 and 4.2 references samples templates 4.1, deleting 4.1 will delete the templates required by 4.2).
I wonder if we should hardcode this specific package id, but it's not great...

Worst case it's not a blocking/critical issue because GameStudio and/or Visual Studio will restore nuget packages on next launch.

Reply:
If this is the only package like this, then it makes sense to make it an exception. I've created a PR #994 that addresses this problem.

Reply:
Sorry, I was thinking about it again and think it's better to not hardcode.
Reasons:
- This `Samples.Templates` package name might change, this would make details of Stride package spill even more into the launcher (and force an upgrade)
- Also I realized there are other packages like this, such as `Stride.OpenTK`

Anyway, I was thinking that a better way to fix that is that versions will (almost) never collide, since the semver now contains the git height (i.e. `4.0.1.1431` with the 1431 being auto-generated).
As a result, we can simply rewrite the test like `dependencyVersionRange.Contains(package.Version)` to `dependencyVersionRange.MinVersion == package.Version && dependencyVersionRange.IsMinInclusive && dependencyVersionRange.MaxVersion == null` ?

Reply:
But we would still end up with the same problem. Imagine the hypothetical scenario like this:
* `MainPackage.1.0` depends on `DependencyPackage.1.0` (minimum version)
* `MainPackage.2.0` depends on `DependencyPackage.1.0` (minimum version)

In this case, the removal of `MainPackage.1.0` would end up deleting `DependencyPackage.1.0`, but it is still needed by `MainPackage.2.0`.

Reply:
Better solution #995 - On launcher start or on GameStudio version install/update/uninstall, all unique dependency versions are enumerated for every installed GameStudio package version. The set is compared to the previous set content, and if previous set contains a version that is no longer present in the new set, only then it is removed.